src/app/pages/visitsCalendar/visitsCalendar.component.ts

```diff
@@ -31,15 +34,20 @@ export class VisitsCalendarComponent implements OnInit {
     public id: string;
 
     constructor(private route: ActivatedRoute,
-                private doctorService: DoctorService,
-                private patientService: PatientService,
-                private appointmentService: AppointmentService,
-                private timeSlotService: TimeSlotService,
-                private authService: AuthenticationService,
-                private router: Router) {
-
+        private doctorService: DoctorService,
+        private patientService: PatientService,
+        private appointmentService: AppointmentService,
+        private timeSlotService: TimeSlotService,
+        private authService: AuthenticationService,
+        private router: Router) {
+
+        this.createTimeSlotComponent = new CreateTimeslotComponent(router, doctorService)
```

Comment:
to nie powinno się jakoś samo magicznie robić?

Reply:
TimeSlotComponent to element GUIowy, wstrzykiwanie elementów GUIowych wydaje mi się spaczone.